lib/posticle/src/grammar.pest

```diff
@@ -0,0 +1,121 @@
+    | ^"resource" | ^"prospero" | ^"ms-visio" | ^"facetime" | ^"diaspora" | ^"platform" | ^"example"
+    | ^"bitcoin" | ^"keyparc" | ^"appdata" | ^"session" | ^"ms-help" | ^"wyciwyg" | ^"coap+ws"
+    | ^"service" | ^"z39.50s" | ^"v-event" | ^"spotify" | ^"message" | ^"onenote" | ^"mupdate"
+    | ^"ms-word" | ^"content" | ^"z39.50r" | ^"beshare" | ^"mongodb" | ^"mailto" | ^"webcal"
+    | ^"lastfm" | ^"jabber" | ^"gopher" | ^"z39.50" | ^"ms-spd" | ^"finger" | ^"mumble" | ^"pkcs11"
+    | ^"market" | ^"rediss" | ^"reload" | ^"magnet" | ^"chrome" | ^"soldat" | ^"spiffe" | ^"callto"
+    | ^"submit" | ^"telnet" | ^"things" | ^"barion" | ^"tn3270" | ^"unreal" | ^"ut2004" | ^"rsync"
+    | ^"ldaps" | ^"ymsgr" | ^"rtmfp" | ^"rtsps" | ^"rtspu" | ^"conti" | ^"coaps" | ^"shttp" | ^"sieve"
+    | ^"skype" | ^"snews" | ^"about" | ^"hxxps" | ^"msnim" | ^"msrps" | ^"steam" | ^"stuns" | ^"https"
+    | ^"notes" | ^"gtalk" | ^"graph" | ^"modem" | ^"turns" | ^"query" | ^"redis" | ^"vemmi" | ^"proxy"
+    | ^"xfire" | ^"ldap" | ^"hxxp" | ^"psyc" | ^"xmpp" | ^"maps" | ^"http" | ^"hcap" | ^"h323"
+    | ^"crid" | ^"lvlt" | ^"itms" | ^"rtmp" | ^"rtsp" | ^"iris" | ^"ircs" | ^"irc6" | ^"ipps"
+    | ^"coap" | ^"sftp" | ^"fish" | ^"info" | ^"sips" | ^"file" | ^"smtp" | ^"feed" | ^"snmp"
+    | ^"elsi" | ^"ed2k" | ^"imap" | ^"msrp" | ^"aaas" | ^"stun" | ^"mtqp" | ^"pwid" | ^"bolo"
+    | ^"blob" | ^"dntp" | ^"tftp" | ^"news" | ^"nntp" | ^"icon" | ^"tool" | ^"turn" | ^"dict"
+    | ^"acap" | ^"pack" | ^"palm" | ^"acct" | ^"icap" | ^"wais" | ^"data" | ^"wpid" | ^"pres"
+    | ^"wtai" | ^"xcon" | ^"jms" | ^"ham" | ^"smb" | ^"sms" | ^"dav" | ^"fax" | ^"jar" | ^"pop"
+    | ^"cap" | ^"irc" | ^"cvs" | ^"dvb" | ^"ssh" | ^"dtn" | ^"mid" | ^"iax" | ^"ipp" | ^"svn"
+    | ^"tag" | ^"git" | ^"tel" | ^"ipn" | ^"geo" | ^"res" | ^"mvn" | ^"rmi" | ^"tip" | ^"ftp"
+    | ^"nfs" | ^"xri" | ^"nih" | ^"moz" | ^"udp" | ^"mms" | ^"urn" | ^"ocf" | ^"oid" | ^"dns"
+    | ^"apt" | ^"aim" | ^"vnc" | ^"afs" | ^"dis" | ^"sgn" | ^"did" | ^"hcp" | ^"wss" | ^"cid"
+    | ^"afp" | ^"sip" | ^"acr" | ^"aaa" | ^"ws" | ^"aw" | ^"tv" | ^"qb" | ^"ni" | ^"gg" | ^"go"
+    | ^"im") ~ "://"
+}
+
+url_domain = _{
+    (!("/" | "?" | "&" | "#" | space | terminator+ ~ (space | EOI)) ~ ANY)+
+}
+
+url_tail = _{
+    ("/" | ("?" | "&" | "#") ~ !(space | EOI))
+    ~ (!(space | terminator+ ~ (space | EOI)) ~ ANY)*
+}
+
+terminator = _{
+    "(" | ")" | "?" | "!" | "*" | "'" | ";" | ":" | "=" | "," | "." | "$" | "%" | "[" | "]"
+    | "~" | "&" | "|" | "@" | "\"" | "<" | ">"
+}
+
+punctuation = _{
+    "-" | "_" | "!" | "\"" | "#" | "$" | "%" | "&" | "'" | "(" | ")" | "*" | "+" | "," | "." | "/" | ":" | ";" | "<" | "=" | ">" | "?" | "@" | "[" | "]" | "^" | "`" | "{" | "|" | "}" | "~"
+}
+
+space = _{
+    " " | "\r" | "\n" | "\t" | '\u{09}'..'\u{0D}' | "\u{20}" | "\u{85}" | "\u{A0}" | "\u{1680}"
+    | "\u{180E}" | '\u{2000}'..'\u{200A}' | "\u{2028}" | "\u{2029}" | "\u{202F}" | "\u{205F}"
```

Comment:
missing U+200B ZERO WIDTH SPACE

Reply:
Bug copied from the regex implementation, will fix.